### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by SLS-BLN**
This is a major update of the Capstone Project. Routing is now established all over the place. Single blog posts and single video posts are dynamically generated with hooks (useNavigate, useParam). I also added some video content for demonstration purposes. Last but not least a Hamburger Menu was integrated.

### CODE REVIEW

File: src/components/Blog/Blog.js [hunk 1]
```diff
@@ -28,6 +37,10 @@ export default function Blog({style}) {
               )
             )}
           </div>
+          <BackButton className="tag-and-back" onClick={() => navigate(-1)}>
```
Comment (PReichetanz) [inline src/components/Blog/Blog.js:40]: I think you don't need the className here and anywhere else.
```suggestion
          <BackButton onClick={() => navigate(-1)}>
```

Comment (PReichetanz) [reply src/components/Blog/Blog.js:40]: Well, I see why you do it, but it still seems strange to me to mix styled components with the className attribute.

File: src/components/Blog/Blog.js [hunk 2]
```diff
@@ -28,6 +37,10 @@ export default function Blog({style}) {
...
+            Zurück
+          </BackButton>
+          <div className="space"></div>
```
Comment (PReichetanz) [inline src/components/Blog/Blog.js:43]: Here, you should rather use padding or margin on the BackButton or the Container in order to create a sufficient gap.

File: src/components/Blog/BlogBox.js [hunk 3]
```diff
@@ -49,13 +60,14 @@ const Box = styled.article`
...
     text-overflow: ellipsis;
+    padding: 0.5rem 0;
   }
 
   p {
     font-size: 1.2rem;
     line-height: 1.2;
     display: -webkit-box; /* stylelint-disable-line value-no-vendor-prefix  */
-    -webkit-line-clamp: var(--line-clamp, 6); /* stylelint-disable-line property-no-vendor-prefix */
+    -webkit-line-clamp: var(--line-clamp, 5); /* stylelint-disable-line property-no-vendor-prefix */
```
Comment (PReichetanz) [inline src/components/Blog/BlogBox.js:70]: If you want to disable the stylelint rule globally, try to add it to the `stylelintrc.json`.

File: src/components/Video/Video.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import styled from 'styled-components/macro';
+import {videoEntries} from './VideoData';
+import {useParams} from 'react-router-dom';
+import {useNavigate} from 'react-router-dom';
+import ReactPlayer from 'react-player/vimeo';
+
+export default function Video({style}) {
+  const navigate = useNavigate();
+  let {slug} = useParams();
+  const obj = videoEntries.filter(item => item.slug === slug);
```
Comment (PReichetanz) [inline src/components/Video/Video.js:10]: I would reconsider the naming of `obj`, it's way too general.

File: src/components/Video/Video.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                <ReactPlayer url={video.videoUrl} width="640" height="480" controls={true} playsinline />
+              </div>
+            </VideoClip>
+            <div className="description">
+              <p className="tag">{video.tag} </p>
+              <h1>{video.titleShort}</h1>
+              <p>{video.description}</p>
+            </div>
+          </Hero>
+          <BackButton className="tag-and-back" onClick={() => navigate(-1)}>
```
Comment (PReichetanz) [inline src/components/Video/Video.js:28]: I don't see why you need this className attribute.

File: src/components/Video/VideoData.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+  {
+    id: nanoid(),
+    title: '10 Sekunden - bloss nicht weniger',
+    titleShort: '10 Sekunden',
+    slug: '10-sekunden-bloss-nicht-weniger',
+    videoUrl: 'https://vimeo.com/737176102',
+    posterFrameUrl: '',
+    tag: 'Dreh',
+    description:
+      'Wenn man vom Fotogarfieren kommt, ist Filmen am Anfang sehr schwer. Der Fehler, den die meisten machen: zu kurze Szenen drehen. Egal wie schwer es fällt, jedes Bild sollte 10 Sekunden lang sein. Man kann am Anfang auch mitzählen',
```
Comment (PReichetanz) [inline src/components/Video/VideoData.js:33]:
```suggestion
      'Wenn man vom Fotografieren kommt, ist Filmen am Anfang sehr schwer. Der Fehler, den die meisten machen: zu kurze Szenen drehen. Egal wie schwer es fällt, jedes Bild sollte 10 Sekunden lang sein. Man kann am Anfang auch mitzählen',
```

File: src/components/Video/VideoData.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+  {
+    id: nanoid(),
+    title: 'Es geht auch ohne Drohne',
+    titleShort: 'Filmen ohne Drohne',
+    slug: 'es-geht-auch-ohne-drohne',
+    videoUrl: 'https://vimeo.com/737175602',
+    posterFrameUrl: '',
+    tag: 'Drohnen',
+    description:
+      'Bilder von Drohnen sind cool, keine Frage. Aber sie kosten auch Zeit. Unter einer halben Stunde geht nichts. Manchmal reicht es, die Kamera aus dem Fenster zu halten und ein Bild zu machen. Das geht nämlich in ein paar Sekunden, wenn man kein Stativ aufauen muss.',
```
Comment (PReichetanz) [inline src/components/Video/VideoData.js:73]:
```suggestion
      'Bilder von Drohnen sind cool, keine Frage. Aber sie kosten auch Zeit. Unter einer halben Stunde geht nichts. Manchmal reicht es, die Kamera aus dem Fenster zu halten und ein Bild zu machen. Das geht nämlich in ein paar Sekunden, wenn man kein Stativ aufbauen muss.',
```

File: src/components/Video/VideoData.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+  {
+    id: nanoid(),
+    title: 'Nahaufnahme sind das Salz in der Suppe',
+    titleShort: 'Nahaufnahme',
+    slug: 'nahaufnahme-sind-das-salz-in-der-suppe',
+    videoUrl: 'https://vimeo.com/737175293',
+    posterFrameUrl: '',
+    tag: 'Dreh',
+    description:
+      'Nähe erzeugt Emotionen. Das ist der eigentich "Trick" in Sachen Detailaufnahmen. Doch wie nah soll man an ein Objekt gehen - vor allem, wenn man ein lebendiges Objekt vor der Linse hat.',
```
Comment (PReichetanz) [inline src/components/Video/VideoData.js:93]:
```suggestion
      'Nähe erzeugt Emotionen. Das ist der eigentiche "Trick" in Sachen Detailaufnahmen. Doch wie nah soll man an ein Objekt gehen - vor allem, wenn man ein lebendiges Objekt vor der Linse hat?',
```

File: src/components/Video/VideoData.js [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  {
+    id: nanoid(),
+    title: 'Filmen in der Industrieproduktion',
+    titleShort: 'Industriefilm',
+    slug: 'filmen-in-der-industrie-produktion',
+    videoUrl: 'https://vimeo.com/737174844',
+    posterFrameUrl: '',
+    tag: 'Dreh',
+    description:
+      'Eigentlich ist nichts ist so einfach, wie in einer Industrieprodutkion zu drehen - wenn man die Abläufe kennt. Vieles wiederholt sich. Und dass sogar alle paar Sekunden. Aber das war es auch schon in Sachen "alles einfach". Die Realität sieht anders aus.',
```
Comment (PReichetanz) [inline src/components/Video/VideoData.js:113]:
```suggestion
      'Eigentlich ist nichts ist so einfach, wie in einer Industrieprodutkion zu drehen - wenn man die Abläufe kennt. Vieles wiederholt sich. Und das sogar alle paar Sekunden. Aber das war es auch schon in Sachen "alles einfach". Die Realität sieht anders aus.',
```

File: src/routes/blogpage.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import styled from 'styled-components/macro';
+
+export default function BlogMain() {
+  return (
+    <Section>
+      {blogEntries.map((blog, index) => {
+        return (
+          <div key={index}>
+            <BlogBox blog={blog} />
+          </div>
```
Comment (PReichetanz) [inline src/routes/blogpage.js:12]: Do you really need the `div` here? You could pass the `key` to the BlogBox as well.

File: src/routes/home.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+        <p>
+          {blog.lead} {blog.text.at(0)}
+        </p>
+        <MoreButton
+          onClick={() => {
+            navigate(`/blog/${blog.slug}`);
+          }}
+        >
+          Mehr dazu ...
+        </MoreButton>
```
Comment (PReichetanz) [inline src/routes/home.js:27]: I think you should add an A11y attribute in order to get an appropriate screen reader text (I found this somewhere else too, but did not comment on it).

File: src/routes/video.js [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+                <div style={{'--aspect-ratio': '16 / 9'}}>
+                  <ReactPlayer url={video.videoUrl} width="640" height="480" controls={true} playsinline />
+                </div>
+                <MoreButtonPlain
+                  onClick={() => {
+                    navigate(`/video/${video.slug}`);
+                  }}
+                >
+                  Mehr dazu ...
+                </MoreButtonPlain>
```
Comment (PReichetanz) [inline src/routes/video.js:42]: aria-label or similar here as well.

File: src/routes/video.js [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+                <div style={{'--aspect-ratio': '16 / 9'}}>
+                  <ReactPlayer url={video.videoUrl} width="640" height="480" controls="true" playsinline />
+                </div>
+                <MoreButtonPlain
+                  onClick={() => {
+                    navigate(`/video/${video.slug}`);
+                  }}
+                >
+                  Mehr dazu ...
+                </MoreButtonPlain>
```
Comment (PReichetanz) [inline src/routes/video.js:65]: aria-label

**Review by PReichetanz — COMMENTED** [review]
Nice work, but the PR is way too long in order to get a high-quality review. Please consider my remarks and make sure that your next PRs are smaller.

File: src/components/Blog/Blog.js (hunk 2 above)
Comment (SLS-BLN) [reply src/components/Blog/Blog.js:43]: Margin didn't work on the parent "Container" but padding does the job!

File: src/components/Blog/BlogBox.js (hunk 3 above)
Comment (SLS-BLN) [reply src/components/Blog/BlogBox.js:70]: That advice came from a Github discussion. I tried to add a rule in `styleintrc.json` but failed (again).

File: src/components/Video/VideoData.js (hunk 9 above)
Comment (SLS-BLN) [reply src/components/Video/VideoData.js:113]: That is a bitter spelling error. Thank you for looking closely.

File: src/components/Blog/Blog.js [hunk 14]
```diff
@@ -7,13 +7,13 @@ import {useNavigate} from 'react-router-dom';
 export default function Blog() {
   const navigate = useNavigate();
   let {slug} = useParams();
-  const obj = blogEntries.filter(item => item.slug === slug);
+  const blogPost = blogEntries.filter(item => item.slug === slug);
 
   return (
     <>
-      {obj.map(blog => (
+      {blogPost.map(blog => (
```
Comment (PReichetanz) [inline src/components/Blog/Blog.js:14]: Wouldn't it be better to use a plural here instead of blogPost?

File: src/components/Video/Video.js [hunk 15]
```diff
@@ -7,11 +7,11 @@ import ReactPlayer from 'react-player/vimeo';
 export default function Video({style}) {
   const navigate = useNavigate();
   let {slug} = useParams();
-  const obj = videoEntries.filter(item => item.slug === slug);
+  const videoPost = videoEntries.filter(item => item.slug === slug);
 
   return (
     <>
-      {obj.map(video => (
+      {videoPost.map(video => (
```
Comment (PReichetanz) [inline src/components/Video/Video.js:14]:
```suggestion
      {videoPosts.map(video => (
```

File: src/components/Video/Video.js [hunk 16]
```diff
@@ -7,11 +7,11 @@ import ReactPlayer from 'react-player/vimeo';
 export default function Video({style}) {
   const navigate = useNavigate();
   let {slug} = useParams();
-  const obj = videoEntries.filter(item => item.slug === slug);
+  const videoPost = videoEntries.filter(item => item.slug === slug);
```
Comment (PReichetanz) [inline src/components/Video/Video.js:10]:
```suggestion
  const videoPosts = videoEntries.filter(item => item.slug === slug);
```

File: src/routes/home.js [hunk 17]
```diff
@@ -19,6 +19,8 @@ export default function Home(style) {
           {blog.lead} {blog.text.at(0)}
         </p>
         <MoreButton
+          role="navigation"
```
Comment (PReichetanz) [inline src/routes/home.js:22]: The `MoreButton` is a button, not a navigation (or a group of navigational elements), so you cannot set the `role="navigation"`.

File: src/routes/video.js [hunk 18]
```diff
@@ -34,6 +34,8 @@ export default function VideoMain() {
                   <ReactPlayer url={video.videoUrl} width="640" height="480" controls={true} playsinline />
                 </div>
                 <MoreButtonPlain
+                  role="navigation"
```
Comment (PReichetanz) [inline src/routes/video.js:37]: Same here with the role attribute.

**Review by PReichetanz — COMMENTED** [review]
Looks good, I left some remarks regarding the latest changes.

**Pull request merged**